go/vt/sqlparser/sql.y

```diff
@@ -317,15 +319,33 @@ insert_or_replace:
   }
 
 update_statement:
-  UPDATE comment_opt aliased_table_name SET update_list where_expression_opt order_by_opt limit_opt
+  UPDATE comment_opt table_references SET update_list where_expression_opt order_by_opt limit_opt
   {
-    $$ = &Update{Comments: Comments($2), Table: $3, Exprs: $5, Where: NewWhere(WhereStr, $6), OrderBy: $7, Limit: $8}
+    $$ = &Update{Comments: Comments($2), TableExprs: $3, Exprs: $5, Where: NewWhere(WhereStr, $6), OrderBy: $7, Limit: $8}
   }
 
 delete_statement:
   DELETE comment_opt FROM table_name where_expression_opt order_by_opt limit_opt
   {
-    $$ = &Delete{Comments: Comments($2), Table: $4, Where: NewWhere(WhereStr, $5), OrderBy: $6, Limit: $7}
+    $$ = &Delete{Comments: Comments($2), TableExprs:  TableExprs{&AliasedTableExpr{Expr:$4}}, Where: NewWhere(WhereStr, $5), OrderBy: $6, Limit: $7}
+  }
+| DELETE comment_opt table_name_list from_or_using table_references where_expression_opt
```

Comment:
Sorry. Missed that part. It's weird syntax. I presume the `from` clause is used to identify rows, but the table list tells you the ones from which the rows have to be dropped.

If this is the case, the table list has to be a subset of what's in the `from` clause. Then the same processTableExprs on the `from` clause should work.